core/dbt/context/providers.py

```diff
@@ -1367,20 +1367,26 @@ def post_hooks(self) -> List[Dict[str, Any]]:
     @contextproperty
     def sql(self) -> Optional[str]:
         # only doing this in sql model for backward compatible
-        if (
-            getattr(self.model, "extra_ctes_injected", None)
-            and self.model.language == ModelLanguage.sql  # type: ignore[union-attr]
-        ):
-            # TODO CT-211
-            return self.model.compiled_code  # type: ignore[union-attr]
-        return None
+        if self.model.language == ModelLanguage.sql:  # type: ignore[union-attr]
+            if getattr(self.model, "defer_relation", None):
```

Comment:
Can we leave some comment here about this logic for future us? You can maybe find the clue in git history but they can also be tricky to find. So I prefer direct comment